crypto/poly1305/poly1305.c

```diff
@@ -1,142 +0,0 @@
-/* Copyright (c) 2014, Google Inc.
- *
- * Permission to use, copy, modify, and/or distribute this software for any
- * purpose with or without fee is hereby granted, provided that the above
- * copyright notice and this permission notice appear in all copies.
- *
- * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
- * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
- * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
- * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
- * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
- * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
- * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. */
-
-/* This implementation of poly1305 is by Andrew Moon
- * (https://github.com/floodyberry/poly1305-donna) and released as public
- * domain. */
-
-#include "internal.h"
-
-#include <assert.h>
-#include <string.h>
-
-#include <openssl/type_check.h>
-
-#include "../internal.h"
-
-
-#define POLY1305_BLOCK_STATE_SIZE 192
-
-
-#if defined(OPENSSL_X86)
-/* See comment above |_poly1305_init_sse2| in poly1305-x86.pl. */
-OPENSSL_COMPILE_ASSERT(POLY1305_BLOCK_STATE_SIZE >= 4 * (5 + 1 + 4 + 2 + 4 * 9),
-                       poly1305_block_state_too_small);
-#elif defined(OPENSSL_X86_64)
-/* See comment above |__poly1305_block| in poly1305-x86_64.pl. */
-OPENSSL_COMPILE_ASSERT(POLY1305_BLOCK_STATE_SIZE >=
-                           4 * (5 + 1 + 2 * 2 + 2 + 4 * 9),
-                       poly1305_block_state_too_small);
-#elif defined(OPENSSL_ARM)
-/* TODO(davidben): Figure out the layout of the struct. For now,
- * |POLY1305_BLOCK_STATE_SIZE| is taken from OpenSSL. */
-#elif defined(OPENSSL_AARCH64)
-/* TODO(davidben): Figure out the layout of the struct. For now,
- * |POLY1305_BLOCK_STATE_SIZE| is taken from OpenSSL. */
-#endif
-
-int GFp_poly1305_init_asm(void *ctx, const uint8_t key[16], void *out_func);
-void GFp_poly1305_blocks(void *ctx, const uint8_t *in, size_t len,
-                         uint32_t padbit);
-void GFp_poly1305_emit(void *ctx, uint8_t mac[16], const uint32_t nonce[4]);
-
-struct poly1305_state_st {
-  alignas(8) uint8_t opaque[POLY1305_BLOCK_STATE_SIZE];
-  uint32_t nonce[4];
```

Comment:
Note that since `opaque` is 8-byte aligned and its size is a multiple of 8, `nonce` is implicitly 8-byte aligned too. Did you check the asm code to see if it expects `nonce` to be aligned?